pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -656,10 +656,18 @@ func (service *SecurityPolicyService) deleteSecurityPolicy(obj interface{}, isVp
 	finalSecurityPolicyCopy := *nsxSecurityPolicy
 	finalSecurityPolicyCopy.Rules = nsxSecurityPolicy.Rules
 
-	if isVpcEnabled(service) || isVpcCleanup {
-		vpcInfo, err := service.getVpcInfo(spNameSpace)
-		if err != nil {
-			return err
+	if isVpcEnabled(service) || isGcOrVpcCleanup {
```

Comment:
Perhaps Gc can't gurantee the Path must synced, we'd better to move it as we discussed.

Reply:
Agreed, I revert changes for GC part.